In [1]:
import pandas as pd

df = pd.read_csv("parsed.csv")
df.head()

,alert,cdi,code,detail,dmin,felt,gap,ids,mag,magType,...,status,time,title,tsunami,type,types,tz,updated,url,parsed_place
0,NaN,NaN,37389218,https://earthquake.usgs.gov/fdsnws/event/1/que...,0.008693,NaN,85.0,",ci37389218,",1.35,ml,...,automatic,1539475168010,"M 1.4 - 9km NE of Aguanga, CA",0,earthquake,",geoserve,nearby-cities,origin,phase-data,",-480.0,1539475395144,https://earthquake.usgs.gov/earthquakes/eventp...,California
1,NaN,NaN,37389202,https://earthquake.usgs.gov/fdsnws/event/1/que...,0.020030,NaN,79.0,",ci37389202,",1.29,ml,...,automatic,1539475129610,"M 1.3 - 9km NE of Aguanga, CA",0,earthquake,",geoserve,nearby-cities,origin,phase-data,",-480.0,1539475253925,https://earthquake.usgs.gov/earthquakes/eventp...,California
2,NaN,4.4,37389194,https://earthquake.usgs.gov/fdsnws/event/1/que...,0.021370,28.0,21.0,",ci37389194,",3.42,ml,...,automatic,1539475062610,"M 3.4 - 8km NE of Aguanga, CA",0,earthquake,",dyfi,focal-mechanism,geoserve,nearby-cities,o...",-480.0,1539536756176,https://earthquake.usgs.gov/earthquakes/eventp...,California
3,NaN,NaN,37389186,https://earthquake.usgs.gov/fdsnws/event/1/que...,0.026180,NaN,39.0,",ci37389186,",0.44,ml,...,automatic,1539474978070,"M 0.4 - 9km NE of Aguanga, CA",0,earthquake,",geoserve,nearby-cities,origin,phase-data,",-480.0,1539475196167,https://earthquake.usgs.gov/earthquakes/eventp...,California
4,NaN,NaN,73096941,https://earthquake.usgs.gov/fdsnws/event/1/que...,0.077990,NaN,192.0,",nc73096941,",2.16,md,...,automatic,1539474716050,"M 2.2 - 10km NW of Avenal, CA",0,earthquake,",geoserve,nearby-cities,origin,phase-data,scit...",-480.0,1539477547926,https://earthquake.usgs.gov/earthquakes/eventp...,California


## a

In [4]:
# a
japan_mb = df[(df["parsed_place"] == "Japan") & (df["magType"] == "mb")]
japan_mb[["parsed_place", "magType", "mag"]].head()

,parsed_place,magType,mag
67,Japan,mb,4.6
713,Japan,mb,4.7
1124,Japan,mb,4.6
1309,Japan,mb,4.5
1398,Japan,mb,4.4


In [5]:
japan_mb["mag"].quantile(0.95)

np.float64(4.9)

The 95th percentile of mb earthquake magnitude in Japan is 4.9.

## b

In [6]:
indonesia = df[df["parsed_place"] == "Indonesia"]
indonesia[["parsed_place", "tsunami"]].head()

,parsed_place,tsunami
9,Indonesia,0
13,Indonesia,0
180,Indonesia,0
421,Indonesia,0
660,Indonesia,0


In [7]:
indonesia["tsunami"].value_counts()

,count
tsunami,
0,113
1,34


In [8]:
34 / (113 + 34) *100

23.12925170068027

23.13% of the earthquakes in Indonesia were coupled with tsunamis.

## c

In [10]:
nevada = df[df["parsed_place"] == "Nevada"]
nevada.describe().round(2)

,cdi,dmin,felt,gap,mag,mmi,nst,rms,sig,time,tsunami,tz,updated
count,15.00,681.00,15.00,681.00,681.0,1.00,681.00,681.00,681.00,6.810000e+02,681.0,681.0,6.810000e+02
mean,2.44,0.17,2.40,153.67,0.5,2.84,12.62,0.15,10.97,1.538314e+12,0.0,-480.0,1.538402e+12
std,0.50,0.17,4.63,68.74,0.7,NaN,9.87,0.08,19.61,5.965637e+08,0.0,0.0,6.010951e+08
min,2.00,0.00,1.00,29.14,-0.5,2.84,3.00,0.00,0.00,1.537247e+12,0.0,-480.0,1.537307e+12
25%,2.00,0.05,1.00,97.38,-0.1,2.84,6.00,0.11,0.00,1.537854e+12,0.0,-480.0,1.537928e+12
50%,2.20,0.11,1.00,149.14,0.4,2.84,10.00,0.15,2.00,1.538280e+12,0.0,-480.0,1.538428e+12
75%,2.90,0.23,1.00,199.72,0.9,2.84,16.00,0.19,12.00,1.538821e+12,0.0,-480.0,1.538878e+12
max,3.30,1.41,19.00,355.91,2.9,2.84,61.00,0.86,129.00,1.539461e+12,0.0,-480.0,1.539483e+12


## d

In [12]:
df["parsed_place"].value_counts().head(20)

,count
parsed_place,
Alaska,3665
California,2861
Nevada,681
Hawaii,367
Puerto Rico,216
Montana,204
Washington,157
Indonesia,147
Idaho,71


In [13]:
ring_locations = ["Bolivia", "Chile", "Ecuador", "Peru", "Costa Rica",
    "Guatemala", "Mexico", "Japan", "Philippines",
    "Indonesia", "New Zealand", "Canada", "Fiji",
    "Alaska", "Washington", "California", "Russia",
    "Taiwan", "Tonga", "Kermadec Islands"]

In [14]:
df.loc[df["parsed_place"].str.contains("Antarctic", case=False, na=False),
    "parsed_place"].value_counts()

,count
parsed_place,
Pacific-Antarctic Ridge,3
Western Indian-Antarctic Ridge,1


In [15]:
df["ring_of_fire"] = (df["parsed_place"].isin(ring_locations)
    | df["parsed_place"].str.contains("Antarctic", case=False, na=False))

df[["parsed_place", "ring_of_fire"]].head()

,parsed_place,ring_of_fire
0,California,True
1,California,True
2,California,True
3,California,True
4,California,True


In [16]:
df.loc[df["parsed_place"].isin(["Mexico", "New Mexico"]),
    ["parsed_place", "ring_of_fire"]].drop_duplicates()

,parsed_place,ring_of_fire
104,Mexico,True
9016,New Mexico,False


## e

In [17]:
df["ring_of_fire"].value_counts()

,count
ring_of_fire,
True,7188
False,2144


There are 7188 earthquakes in the Ring of Fire locations and 2144 out of the locations.

## f

In [18]:
df.loc[df["ring_of_fire"], "tsunami"].sum()

np.int64(45)

The tsunami count along the Ring of Fire is 45.